source/general_definitions.hpp

```diff
@@ -103,10 +103,8 @@ class Basis {
      */
     virtual DynMatrix<double> GetMinv(const uint p) = 0;
 
-    template <typename InputArrayType>
-    decltype(auto) ProjectBasisToLinear(const InputArrayType& u);
-    template <typename InputArrayType>
-    decltype(auto) ProjectLinearToBasis(const uint ndof, const InputArrayType& u_lin);
+    virtual DynMatrix<double> ProjectBasisToLinear(const DynMatrix<double>& u)                      = 0;
```

Comment:
Why do we need this?

I'd prefer to keep the expression templates intact as these projections get used in slope limiting and wetting and drying.

Reply:
If you look at previous implementation, I had to create a new DynMatrix with projected values and return. So there are no expression templates going on in these function. I though about a way to use expression templates and stuff. I need to store a matrix in basis class and I though maybe that's not the best option.